sspl_test/alerts/self_hw/test_node_bmc.py

```diff
@@ -0,0 +1,169 @@
+                print("UNEXPECTED BMC CHANNEL TYPE FOUND.")
+                print("Expected: %s" % expected_channel)
+                print("Found: %s" % channel_found)
+        else:
+            res_err = res_err.decode()
+            kcs_errors = ("could not find inband device", "driver timeout")
+            if not any(err for err in kcs_errors if err in res_err):
+                raise Exception("ERROR: %s" % res_err)
+    elif channel_interface == "lan":
+        # Check BMC is accessible through LAN
+        subcommand = "channel info"
+        bmc_ip = Conf.get(GLOBAL_CONF, BMC_IP_KEY)
+        bmc_user = Conf.get(GLOBAL_CONF, BMC_USER_KEY)
+        bmc_secret = Conf.get(GLOBAL_CONF, BMC_SECRET_KEY)
+        bmc_key = Cipher.generate_key(MACHINE_ID, "server_node")
+        bmc_passwd = Cipher.decrypt(
+            bmc_key, bmc_secret.encode("utf-8")).decode("utf-8")
+        cmd = "sudo ipmitool -H %s -U %s -P %s -I lan %s" %(
+            bmc_ip, bmc_user, bmc_passwd, subcommand)
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc != 0:
+            raise Exception("ERROR: %s" % res_err.decode())
+
+def test_chassis_selftest(args):
+    """Check chassis selttestsel is passed."""
+    cmd = "ipmitool chassis selftest"
+    expected_res = "Self Test Results    : passed"
+    res_op, res_err, res_rc = SimpleProcess(cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        if expected_res not in res_op:
+            assert False, res_op
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+
+def test_sensor_availability(args):
+    """Fail if any expected sensor is not detected by ipmitool."""
+    found_all_sensors = True
+    sensors = [
+        "Voltage",
+        "Temperature",
+        "Power Supply",
+        "Drive Slot / Bay",
+        "Fan"
+        ]
+    # Get manufacturer name
+    manufacturer = get_manufacturer_name()
+    for sensor in sensors:
+        cmd = ["ipmitool", "sdr", "type", sensor]
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc == 0:
+            res_op = res_op.decode().replace("\n", "")
+            if not res_op:
+                found_all_sensors = False
+                print(
+                    "'%s' sensor is not seen in %s node server." % (
+                        sensor, manufacturer))
+        else:
+            raise Exception("ERROR: %s" % res_err.decode())
+    assert(found_all_sensors == True)
+
+def test_ipmitool_sel_accessibility(args):
+    """Check sel list is accessible."""
+    sel_command = "ipmitool sel list"
+    _, res_err, res_rc = SimpleProcess(sel_command).run()
+    if res_rc != 0:
+        res_err = res_err.decode()
+        assert False, "CMD failure: %s" % res_err
+
+
+test_list = [
+    test_bmc_config,
+    test_bmc_firmware_version,
+    test_bmc_is_accessible,
+    test_chassis_selftest,
+    test_sensor_availability,
+    test_ipmitool_sel_accessibility
```

Comment:
How can we miss ipmi version check itself. 
test_ipmi_version can be other test

>  ipmitool bmc info
Device ID                 : 34
Device Revision           : 1
Firmware Revision         : 1.93
**IPMI Version              : 2.0**

Reply:
Done.
Addressed in PR #402